Unable to build rr on Fedora Rawhide

When building rr on Fedora Rawhide, there are a few narrowing conversion errors.

```
CMakeFiles/rr.dir/src/SeccompFilterRewriter.cc.o -c /builddir/build/BUILD/rr-cf5169bb3e29ce9db4a73e26164bec0e92b083fb/src/SeccompFilterRewriter.cc
/builddir/build/BUILD/rr-cf5169bb3e29ce9db4a73e26164bec0e92b083fb/src/Monkeypatcher.cc: In member function 'bool rr::Monkeypatcher::try_patch_vsyscall_caller(rr::RecordTask*, rr::remote_code_ptr)':
/builddir/build/BUILD/rr-cf5169bb3e29ce9db4a73e26164bec0e92b083fb/src/Monkeypatcher.cc:410:10: error: narrowing conversion of '18446744073699065856' from 'long unsigned int' to 'long int' [-Wnarrowing]
  410 |     case 0xffffffffff600000:
      |          ^~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/rr-cf5169bb3e29ce9db4a73e26164bec0e92b083fb/src/Monkeypatcher.cc:413:10: error: narrowing conversion of '18446744073699066880' from 'long unsigned int' to 'long int' [-Wnarrowing]
  413 |     case 0xffffffffff600400:
      |          ^~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/rr-cf5169bb3e29ce9db4a73e26164bec0e92b083fb/src/Monkeypatcher.cc:416:10: error: narrowing conversion of '18446744073699067904' from 'long unsigned int' to 'long int' [-Wnarrowing]
  416 |     case 0xffffffffff600800:
```

--- a/src/Monkeypatcher.cc
+++ b/src/Monkeypatcher.cc
@@ -404,7 +404,7 @@ bool Monkeypatcher::try_patch_vsyscall_caller(RecordTask* t, remote_code_ptr ret
   if (!X64VSyscallEntry::match(bytes, &target_addr)) {
     return false;
   }
-  int64_t target_addr_sext = (int32_t)target_addr;
+  uint64_t target_addr_sext = (uint64_t)(int32_t)target_addr;
   int syscallno = 0;
   switch (target_addr_sext) {
     case 0xffffffffff600000:

